sklearn/manifold/tests/test_isomap.py

```diff
@@ -173,38 +179,37 @@ def test_pipeline_with_nearest_neighbors_transformer():
 
     Xt_chain = est_chain.fit_transform(X)
     Xt_compact = est_compact.fit_transform(X)
-    assert_array_almost_equal(Xt_chain, Xt_compact)
+    assert_allclose(Xt_chain, Xt_compact)
 
     Xt_chain = est_chain.transform(X2)
     Xt_compact = est_compact.transform(X2)
-    assert_array_almost_equal(Xt_chain, Xt_compact)
+    assert_allclose(Xt_chain, Xt_compact)
 
 
-def test_different_metric():
-    # Test that the metric parameters work correctly, and default to euclidean
-    def custom_metric(x1, x2):
-        return np.sqrt(np.sum(x1**2 + x2**2))
-
-    # metric, p, is_euclidean
-    metrics = [
+@pytest.mark.parametrize(
+    "metric, p, is_euclidean",
+    [
         ("euclidean", 2, True),
         ("manhattan", 1, False),
         ("minkowski", 1, False),
         ("minkowski", 2, True),
-        (custom_metric, 2, False),
-    ]
-
+        (lambda x1, x2: np.sqrt(np.sum(x1**2 + x2**2)), 2, False),
+    ],
+)
+def test_different_metric(global_dtype, metric, p, is_euclidean):
+    # Isomap must work on various metric parameters work correctly
+    # and must default to euclidean.
     X, _ = datasets.make_blobs(random_state=0)
-    reference = manifold.Isomap().fit_transform(X)
+    X = X.astype(global_dtype, copy=False)
 
-    for metric, p, is_euclidean in metrics:
-        embedding = manifold.Isomap(metric=metric, p=p).fit_transform(X)
+    reference = manifold.Isomap().fit_transform(X)
+    embedding = manifold.Isomap(metric=metric, p=p).fit_transform(X)
 
-        if is_euclidean:
-            assert_array_almost_equal(embedding, reference)
-        else:
-            with pytest.raises(AssertionError, match="not almost equal"):
-                assert_array_almost_equal(embedding, reference)
+    if is_euclidean:
+        assert_allclose(embedding, reference)
+    else:
+        with pytest.raises(AssertionError, match="Not equal to tolerance rtol=1e-07"):
```

Comment:
This one is easy
```suggestion
        with pytest.raises(AssertionError, match="Not equal to tolerance"):
```

Reply:
Ah let's put it under the carpet? :smile:

Reply:
no it's just that the tolerance depends on the dtype so expecting 1e-7 makes it fails when dtype is float32 :)

Reply:
Yeah, I tried to change the tolerance in the message based on the configuration but had tests failing for another reason then. This LGTM. :)